src/Jackalope/Version/VersionHandler.php

```diff
@@ -245,5 +247,172 @@ private function createFrozenNode(NodeInterface $versionNode, NodeInterface $nod
 
             $frozenNode->setProperty($propertyName, $property->getValue());
         }
+
+        foreach ($node->getNodes() as $childNode) {
+            /** @var NodeInterface $childNode */
+            // TODO also use the commented lines when getDefinition is implemented
+            //$onParentVersion = $childNode->getDefinition()->getOnParentVersion();
```

Comment:
@dbu I would like to also implement the other onParentVersion cases, but the `getDefinition` method of the `Node` is not implemented: https://github.com/jackalope/jackalope/blob/ec1020b8431969708c210c8de88c9a36fd78151f/src/Jackalope/Node.php#L1125

Can this method be almost borrowed from the `Property`?